libparsec/crates/platform_storage/src/native/workspace/mod.rs

```diff
@@ -3,6 +3,7 @@
 // Certificate storage relies on the upper layer to do the actual certification
 // validation work and takes care of handling concurrency issues.
 // Hence no unique violation should occur under normal circumstances here.
+mod sync_pattern;
```

Comment:
I'd rather keep everything in the same file for the moment: having multiple impl of the same structure in different file is error prone (see what is done for `CertificatesOps`/`WorkspaceOps` etc. to split the code while having a single impl)

With your changes added, the file will be ~1000lines which in the high limit, but I think it's still okay given the code there is very simple (basically impl PlatformWorkspaceStorage exposing methods, and a bunch of `db_` function to implement the queries, with each method being very isolated from each other).

Reply:
done